canonical-web-and-design/maas-ui
# fix(ui): domain details cleanup

## Done

- Created separate types for BaseDomain and DomainDetails + added util `isDomainDetails()` 
- Added test factories for domain details and domain resources
- Used `Expanded` pattern we use elsewhere in resource records table
- Moved `AddRecordFields` to `components/RecordFields` as it's used in both add and edit
- Removed `updateDNSResource` action from the edit record form for now - seems like we do actually need that to complete first before calling `updateAddressRecord` so I'll handle that in a followup
- Moved `RecordType` to `enum` file
- Changed `DeleteDomainForm` to `FormikForm` so errors are handled
- Fixed some form validation (TTL must be greater than 1 or be empty i.e. default)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that you can still add and edit domains
- Check that you can still add and edit resource records (but not their names for now)
- Check that you can still delete a domain

## Fixes

Fixes canonical-web-and-design/app-squad#126
